guillotina_amqp/commands/worker.py

```diff
@@ -61,14 +63,28 @@ class WorkerCommand(Command):
     def get_parser(self):
         parser = super().get_parser()
         parser.add_argument('--auto-kill-timeout',
-                            help='How long of no activity before we automatically kill process',
+                            help='How long of no activity before we automatically kill process (in minutes)',
                             type=int, default=-1)
         parser.add_argument('--max-running-tasks',
                             help='Max simultaneous running tasks',
                             type=int, default=None)
+        parser.add_argument('--metrics-server', help='Launch an API server to expose metrics',
+                            default=False, action='store_true')
         return parser
 
-    async def run(self, arguments, settings, app):
+    def run(self, arguments, settings, app):
+        loop = self.get_loop()
+        fut = loop.run_in_executor(None, self.run_worker, arguments, settings, app)
```

Comment:
I think you want to use `asyncio.ensure_future` here only and not with `run_in_executer`.

Also, I think it should be conditional and don't don this way if you are using the metrics server!

Reply:
I tried with ensure_future already, but the worker wasn't picking up new tasks. I will give it another shot :+1: 

It is actually conditional on the parameter `--metrics-server`.